### PR TITLE
Add recipe for mbsolve.

### PR DESCRIPTION
**Pull request by mriesch-tum**
mbsolve is an open-source solver tool for the full wave 1D Maxwell-Bloch equations.

Checklist
- [X] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml" 
- [X] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [X] Source is from official source
- [X] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [x] <s>If static libraries are linked in, the license of the static library is packaged.</s>
- [X] Build number is 0
- [X] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [X] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

TODO:
 - [x] Update source to tagged version (which should be released soon)
 - [x] Revise requirements: is host section required?

### CODE REVIEW

**Comment by mriesch-tum** [comment]
Pull request is ready to review (and ready to merge, as far as I am concerned). 

@conda-forge/help-c-cpp
@conda-forge/help-python

Thanks!

File: recipes/mbsolve/bld.bat [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+cmake ^
+    -G "Ninja" ^
+    -DCMAKE_PREFIX_PATH="%LIBRARY_PREFIX%" ^
+    -DCMAKE_INSTALL_PREFIX="%LIBRARY_PREFIX%" ^
+    -DPYTHON_LIBRARY="%PREFIX%\libs\python%CONDA_PY%.lib" ^
+    -DCMAKE_BUILD_TYPE=Release ^
+    ..
+
+cmake --build . --config Release
+cmake --build . --config Release --target install
```
Comment (scopatz) [inline recipes/mbsolve/bld.bat]: This file needs exit clauses after each line

Comment (mriesch-tum) [reply recipes/mbsolve/bld.bat]: I am not too familiar with Windows batch scripts -- not exactly sure what you mean. Could you point me to an example (recipe), please?

Comment (mriesch-tum) [reply recipes/mbsolve/bld.bat]: I am not too familiar with Windows batch scripts -- not exactly sure what you mean. Could you point me to an example (recipe), please?

Comment (scopatz) [reply recipes/mbsolve/bld.bat]: See this example: https://github.com/conda-forge/make-feedstock/blob/master/recipe/bld.bat#L5

Comment (mriesch-tum) [reply recipes/mbsolve/bld.bat]: Thanks! I have added another commit that addresses the missing clauses.

**Comment by scopatz** [comment]
Thanks!

**Pull request merged**